mteb/models/jina_models.py

```diff
@@ -0,0 +1,235 @@
+    "hau_Latn",
+    "heb_Latn",
+    "hin_Latn",
+    "hin_Deva",
+    "hrv_Latn",
+    "hun_Latn",
+    "hye_Latn",
+    "ind_Latn",
+    "isl_Latn",
+    "ita_Latn",
+    "jpn_Latn",
+    "jav_Latn",
+    "kat_Latn",
+    "kaz_Latn",
+    "khm_Latn",
+    "kan_Latn",
+    "kor_Latn",
+    "kur_Latn",
+    "kir_Latn",
+    "lat_Latn",
+    "lao_Latn",
+    "lit_Latn",
+    "lav_Latn",
+    "mlg_Latn",
+    "mkd_Latn",
+    "mal_Latn",
+    "mon_Latn",
+    "mar_Latn",
+    "msa_Latn",
+    "mya_Latn",
+    "nep_Latn",
+    "nld_Latn",
+    "nob_Latn",
+    "orm_Latn",
+    "ori_Latn",
+    "pan_Latn",
+    "pol_Latn",
+    "pus_Latn",
+    "por_Latn",
+    "ron_Latn",
+    "rus_Latn",
+    "san_Latn",
+    "snd_Latn",
+    "sin_Latn",
+    "slk_Latn",
+    "slv_Latn",
+    "som_Latn",
+    "sqi_Latn",
+    "srp_Latn",
+    "sun_Latn",
+    "swe_Latn",
+    "swa_Latn",
+    "tam_Latn",
+    "tam_Taml",
+    "tel_Latn",
+    "tel_Telu",
+    "tha_Latn",
+    "tgl_Latn",
+    "tur_Latn",
+    "uig_Latn",
+    "ukr_Latn",
+    "urd_Latn",
+    "urd_Arab",
+    "uzb_Latn",
+    "vie_Latn",
+    "xho_Latn",
+    "yid_Latn",
+    "zho_Hant",
+    "zho_Hans",
+]
+
+
+class JinaWrapper(SentenceTransformerWrapper):
+    """following the hf model card documentation."""
+
+    jina_task_to_prompt = {
+        "retrieval.query": "Represent the query for retrieving evidence documents: ",
+        "retrieval.passage": "Represent the document for retrieval: ",
+    }
+
+    def __init__(
+        self,
+        model: str,
+        revision: str | None = None,
+        model_prompts: dict[str, str] | None = None,
+        **kwargs,
+    ) -> None:
+        if CURRENT_SENTENCE_TRANSFORMERS_VERSION < MIN_SENTENCE_TRANSFORMERS_VERSION:
+            raise RuntimeError(
+                f"sentence_transformers version {st_version} is lower than the required version 3.1.0"
+            )
+        try:
+            import einops  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "To use the jina-embeddings-v3 models `einops` is required. Please install it with `pip install einops`."
+            )
+        super().__init__(model, revision, model_prompts, **kwargs)
+
+    def encode(
+        self,
+        sentences: Sequence[str],
+        *,
+        task_name: str,
+        prompt_type: PromptType | None = None,
+        **kwargs: Any,
+    ) -> np.ndarray:
+        """Encodes the given sentences using the encoder.
+
+        Args:
+            sentences: The sentences to encode.
+            task_name: The name of the task. Sentence-transformers uses this to
+                determine which prompt to use from a specified dictionary.
+            prompt_type: The name type of prompt. (query or passage)
+            **kwargs: Additional arguments to pass to the encoder.
+
+            The order of priorities for prompt selection are:
+                1. Composed prompt of task name + prompt type (query or passage)
```

Comment:
is this relevant to have here?

Reply:
Mostly not, I copied this from wrapper

Reply:
let us remove it then (also from the other encode)